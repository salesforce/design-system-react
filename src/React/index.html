<!doctype html public "library">
<meta charset=utf-8/>
<head>
	<title>Selectlist</title>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
</head>
<body>
<script src="/dist/react/build.js"></script>
</body>
</html>