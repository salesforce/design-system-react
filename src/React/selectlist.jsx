//test

var React = require('react');
import {SelectlistCore} from "../SelectlistCore";
var DropdownButton = require('react-bootstrap/lib/DropdownButton');
var MenuItem = require('react-bootstrap/lib/MenuItem');

var DATA = {
	selected: 'None selected',
	id: 'selectlist1',
	items: [
		{ id: 0, name: 'tacos', type: 'mexican' },
		{ id: 1, name: 'burrito', type: 'mexican' },
		{ id: 2, name: 'tostada', type: 'mexican' },
		{ id: 3, name: 'hush puppies', type: 'southern' }
	]
};

var menuItemSelected = (eventKey, href, target)=> {
	DATA.selected = DATA.items[eventKey].name;
};
//Selectlist.handleMenuItemClicked
var menuItems = DATA.items.map( (menuItem) => {
	return <MenuItem eventKey={menuItem.id}>{menuItem.name}</MenuItem>;
});

var Selectlist = React.createClass({
	getInitialState: function() {
		return {
			selectedKey: undefined
		};
	},

	handleMenuItemClicked: function(eventKey, href, target) {
		this.setState({
			selectedKey: eventKey
		});
	},

	menuItems: function() {
		return DATA.items.map( (menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>{menuItem.name}</MenuItem>;
		});
	},

	selected: function() {
		return typeof(this.state.selectedKey) === 'undefined' ? 'None Selected' : DATA.items[this.state.selectedKey].name;
	},

	render () {
		return (
			<div>
				<h1>SelectList</h1>
				<ul>{this.menuItems}</ul>
				<DropdownButton title={this.selected()} key={DATA.id}>{this.menuItems()}</DropdownButton>
			</div>
		);
	}
});

React.render(<Selectlist/>, document.body, () => {});