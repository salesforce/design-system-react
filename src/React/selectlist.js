// SELECTLIST CONTROL

// core
import {Landmark} from '../Landmark';
import {SelectlistCore} from '../SelectlistCore';

// framework specific
import React from 'react';
import DropdownButton from 'react-bootstrap/lib/DropdownButton';
import MenuItem from 'react-bootstrap/lib/MenuItem';

export var Selectlist = React.createClass(Object.assign({}, SelectlistCore, {	
	propTypes: {
		disabled: React.PropTypes.bool,
		selection: React.PropTypes.number,
		collection: React.PropTypes.array
	},
	
	getInitialState () {
		return this.__getInitialState();
	},
		
	menuItems () {
		return this.props.collection.map((menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>
					{menuItem.name}
				</MenuItem>;
		});
	},
	
	componentWillReceiveProps(nextProps) {
		this.__initializeOptions(nextProps);
	},
	
	render () {
		var selection = this.getSelection();
		
		return (
			<DropdownButton className={this.cssClasses.CONTROL} disabled={this.state.disabled} title={selection ? selection.name : 'None selected'} key={this.props.id}>{this.menuItems()}</DropdownButton>
		);
	},
	
	componentWillMount () {
		this.elements = {
			wrapper: {
				toggleClass: function (cssClass, state) {
					Landmark.log(cssClass, state);
				}
			}
		}
		
		this.__constructor(this.props);
	},
	
	handleMenuItemClicked (eventKey, href, target) {
		this.setSelectionByKey('id', eventKey);
	}
}));