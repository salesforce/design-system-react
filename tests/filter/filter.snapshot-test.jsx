import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ReactDOMServer from 'react-dom/server';
import jsBeautify from 'js-beautify';

import Default from '../../examples/filter/default';
import NewFilter from '../../examples/filter/new';
import LockedFilter from '../../examples/filter/locked';
import PermanantFilter from '../../examples/filter/permanant';
import ErrorFilter from '../../examples/filter/error';

test('Filter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<Default />
	));
	expect(domTree).toMatchSnapshot();
});

test('NewFilter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<NewFilter />
	));
	expect(domTree).toMatchSnapshot();
});

test('LockedFilter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<LockedFilter />
	));
	expect(domTree).toMatchSnapshot();
});

test('Permanant Filter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<PermanantFilter />
	));
	expect(domTree).toMatchSnapshot();
});

test('Error Filter Base Snapshot', () => {
	const domTree = toJson(shallow(
		<ErrorFilter />
	));
	expect(domTree).toMatchSnapshot();
});

test('Filter Base with custom className Snapshot', () => {
	const domTree = String(
		jsBeautify.html(ReactDOMServer.renderToStaticMarkup(<Default className="MY_CUSTOM_CLASS_NAME" />), { indent_size: 2 }),
		'utf-8'
	);
	expect(domTree).toMatchSnapshot();
});
