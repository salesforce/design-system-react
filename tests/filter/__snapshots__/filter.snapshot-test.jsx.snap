exports[`test AssistiveText Filter 1`] = `
<SLDSFilter
  align="left"
  assistiveText={
    Object {
      "editFilter": "editFilter-TEST",
      "editFilterHeading": "editFilterHeading-TEST",
      "removeFilter": "removeFilter-TEST",
    }
  }
  id="sample-panel-filtering-show-me"
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;

exports[`test AssistiveText Filter HTML Snapshot 1`] = `
"<div class=\"slds-filters__item slds-grid slds-grid--vertical-align-center\">
  <div class=\"slds-grow\" style=\"display:inline;\"><button class=\"slds-button--reset slds-grow slds-has-blur-focus\" id=\"sample-panel-filtering-show-me\" tabindex=\"0\"><span class=\"slds-assistive-text\">Edit filter:</span><p class=\"slds-text-body--small\">Show Me</p><p>All Products</p></button></div>
  <button
    class=\"slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small\" title=\"Remove Filter: Show Me All Products\"><svg aria-hidden=\"true\" class=\"slds-button__icon slds-button__icon--hint\" viewBox=\"0 0 24 24\" name=\"close\"><path d=\"M14.3 11.7l6-6c.3-.3.3-.7 0-1l-.9-1c-.3-.2-.7-.2-1 0l-6 6.1c-.2.2-.5.2-.7 0l-6-6.1c-.3-.3-.7-.3-1 0l-1 1c-.2.2-.2.7 0 .9l6.1 6.1c.2.2.2.4 0 .6l-6.1 6.1c-.3.3-.3.7 0 1l1 1c.2.2.7.2.9 0l6.1-6.1c.2-.2.4-.2.6 0l6.1 6.1c.2.2.7.2.9 0l1-1c.3-.3.3-.7 0-1l-6-6c-.2-.2-.2-.5 0-.7z\"></path></svg>
    <span
      class=\"slds-assistive-text\">Remove Filter: Show Me All Products</span>
      </button>
</div>"
`;

exports[`test Error Filter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveText={
    Object {
      "editFilter": "Edit filter:",
      "editFilterHeading": "Choose filter criteria",
    }
  }
  id="sample-panel-filtering-show-me"
  isError={true}
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;

exports[`test Filter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveText={
    Object {
      "editFilter": "Edit filter:",
      "editFilterHeading": "Choose filter criteria",
    }
  }
  id="sample-panel-filtering-show-me"
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;

exports[`test Filter Base with custom className Snapshot 1`] = `
"<div class=\"slds-filters__item slds-grid slds-grid--vertical-align-center MY_CUSTOM_CLASS_NAME\">
  <div class=\"slds-grow\" style=\"display:inline;\"><button class=\"slds-button--reset slds-grow slds-has-blur-focus\" id=\"sample-panel-filtering-show-me\" tabindex=\"0\"><span class=\"slds-assistive-text\">Edit filter:</span><p class=\"slds-text-body--small\">Show Me</p><p>All Products</p></button></div>
  <button
    class=\"slds-button slds-button--icon slds-button--icon-bare slds-button--icon-small\" title=\"Remove Filter: Show Me All Products\"><svg aria-hidden=\"true\" class=\"slds-button__icon slds-button__icon--hint\" viewBox=\"0 0 24 24\" name=\"close\"><path d=\"M14.3 11.7l6-6c.3-.3.3-.7 0-1l-.9-1c-.3-.2-.7-.2-1 0l-6 6.1c-.2.2-.5.2-.7 0l-6-6.1c-.3-.3-.7-.3-1 0l-1 1c-.2.2-.2.7 0 .9l6.1 6.1c.2.2.2.4 0 .6l-6.1 6.1c-.3.3-.3.7 0 1l1 1c.2.2.7.2.9 0l6.1-6.1c.2-.2.4-.2.6 0l6.1 6.1c.2.2.7.2.9 0l1-1c.3-.3.3-.7 0-1l-6-6c-.2-.2-.2-.5 0-.7z\"></path></svg>
    <span
      class=\"slds-assistive-text\">Remove Filter: Show Me All Products</span>
      </button>
</div>"
`;

exports[`test LockedFilter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveText={
    Object {
      "editFilter": "Edit filter:",
      "editFilterHeading": "Choose filter criteria",
    }
  }
  id="sample-panel-filtering-show-me"
  isLocked={true}
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;

exports[`test NewFilter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveText={
    Object {
      "editFilter": "Edit filter:",
      "editFilterHeading": "Choose filter criteria",
    }
  }
  id="sample-panel-filtering-show-me"
  isNew={true}
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;

exports[`test Permanant Filter Base Snapshot 1`] = `
<SLDSFilter
  align="left"
  assistiveText={
    Object {
      "editFilter": "Edit filter:",
      "editFilterHeading": "Choose filter criteria",
    }
  }
  id="sample-panel-filtering-show-me"
  isPermanent={true}
  onChange={[Function]}
  onRemove={[Function]}
  predicate="All Products"
  property="Show Me">
  <SLDSMenuPicklist
    checkmark={true}
    inheritTargetWidth={true}
    isInline={true}
    label="Show Me"
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "All Products",
          "value": "all-products",
        },
        Object {
          "label": "All Wackamoles",
          "value": "all-Wackamoles",
        },
      ]
    }
    placeholder="Select record type"
    value="all-products" />
</SLDSFilter>
`;
